ath79: Add support for TP-Link Archer C59v2

kicken opened the pull request:
Adds support for TP-Link Archer C59v2 to the ath79 platform.

Based on changes made in PR #2364

adschm committed:
ath79: move common definitions from Archer C58/C59 to common DTSI

The Archer C58/C59 have redundant LED and MAC address definitions
in their DTS files. This moves them to the parent DTSI file.

The patch already accounts for the upcoming Archer C59 v2.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
Hi, I would actually just copy the device specs and flashing instructions from the ar71xx commit message instead of referring to them.

adrianschmutzler commented:
Note for reviewers:
This will require the same LED migration as proposed in https://patchwork.ozlabs.org/patch/1156538/

However, as long as the referred patch is on hold, I would not require this PR to care about migration.

adrianschmutzler commented on target/linux/ath79/dts/qca9561_tplink_archer-c59-v2.dts:
```diff
@@ -0,0 +1,86 @@
+        linux,default-trigger = "usbport";
+        trigger-sources = <&hub_port>;
+    };
+};
```
Indent in the block above is wrong (spaces instead of tabs).

adrianschmutzler replied:
Done.

adschm reviewed (approved):
Should be ready for merge.
My patch can be used from this source, there is no difference to the version in the earlier separate PR.
@blocktrron

blocktrron commented:
Thanks! Merged to my staging tree.

adrianschmutzler commented:
In your staging tree it looks like the second patch has author-signed-off mismatch in surname ...

blocktrron commented:
Yes, I've missed reopening this PR yesterday.

@kicken Can you fix the commit author for your second patch?

adrianschmutzler commented:
@kicken If your Signed-off is correct, the easiest is:
`git commit --amend --reset-author` 
This will reset the author to your current signed-off.

kicken committed:
ath79: add support for TP-Link Archer C59v2

Add ath79 support for Archer C59v2, previously supported by ar71xx.

TP-Link Archer C59v2 is a dual-band AC1350 router based on
Qualcomm/Atheros QCA9561+QCA9886 chips.

Specification:

- 775/650/258 MHz (CPU/DDR/AHB)
- 128 MB of RAM (DDR2)
- 16 MB of FLASH (SPI NOR)
- 3T3R 2.4 GHz
- 2T2R 5 GHz
- 5x 10/100 Mbps Ethernet
- USB 2.0 port
- UART header on PCB

Flash instruction:
  - via web UI:
    1. Download openwrt-ath79-generic-tplink_archer-c59-v2-squashfs-factory.bin
    2. Login to router and open the Advanced tab
    3. Navigate to System Tools -> Firmware Upgrade
    4. Upload firmware using the Manual Upgrade form
  - via TFTP:
    1. Set PC to fixed ip address 192.168.0.66
    2. Download openwrt-ath79-generic-tplink_archer-c59-v2-squashfs-factory.bin
       and rename it to tp_recovery.bin
    3. Start a tftp server with the file tp_recovery.bin in its root directory
    4. Turn off the router
    5. Press and hold Reset button
    6. Turn on router with the reset button pressed and wait ~15 seconds
    7. Release the reset button and after a short time
       the firmware should be transferred from the tftp server
    8. Wait ~30 second to complete recovery.

Signed-off-by: Keith Maika <keithm@aoeex.com>

kicken commented:
@blocktrron Done.
